app/screens/members_modal/members_modal.tsx

```diff
@@ -0,0 +1,332 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo, useReducer, useRef, useState} from 'react';
+import {MessageDescriptor, useIntl} from 'react-intl';
+import {Keyboard, Platform, StyleSheet, View} from 'react-native';
+import {SafeAreaView} from 'react-native-safe-area-context';
+
+import CompassIcon from '@components/compass_icon';
+import Loading from '@components/loading';
+import Search from '@components/search';
+import {General} from '@constants';
+import {useTheme} from '@context/theme';
+import {debounce} from '@helpers/api/general';
+import useNavButtonPressed from '@hooks/navigation_button_pressed';
+import SelectedUsers from '@screens/members_modal/selected_users';
+import UserList from '@screens/members_modal/user_list';
+import {dismissModal, setButtons} from '@screens/navigation';
+import {changeOpacity, getKeyboardAppearanceFromTheme} from '@utils/theme';
+import {filterProfilesMatchingTerm} from '@utils/user';
+
+const style = StyleSheet.create({
+    container: {
+        flex: 1,
+    },
+    searchBar: {
+        marginLeft: 12,
+        marginRight: Platform.select({ios: 4, default: 12}),
+        marginVertical: 12,
+    },
+});
+
+const ACTION_BUTTON = 'action-button';
+const CLOSE_BUTTON = 'close-dms';
+
+const close = () => {
+    Keyboard.dismiss();
+    dismissModal();
+};
+
+type searchError = {
+    data?: undefined;
+    error: unknown;
+}
+
+type searchSuccess = {
+    data: UserProfile[];
+    error?: undefined;
+}
+
+type getProfilesError = {
+    users?: undefined;
+    error: unknown;
+}
+
+type getProfilesSuccess = {
+    users: UserProfile[];
+    error?: undefined;
+}
+
+type Props = {
+    buttonText: MessageDescriptor;
+    componentId: string;
+    currentUserId: string;
+    getProfiles: () => Promise<getProfilesSuccess | getProfilesError>;
+    maxSelectedUsers: number;
+    page: number;
+    searchUsers: (searchTerm: string) => Promise<searchError | searchSuccess>;
+    selectedIds: {[id: string]: UserProfile};
+    setPage: (page: number) => void;
+    setSelectedIds: (ids: {[id: string]: UserProfile}) => void;
+    onButtonTap: (selectedId?: {[id: string]: boolean}) => Promise<boolean>;
+    teammateNameDisplay: string;
+    tutorialWatched: boolean;
+}
+
+function reduceProfiles(state: UserProfile[], action: {type: 'add'; values?: UserProfile[]}) {
+    if (action.type === 'add' && action.values?.length) {
+        return [...state, ...action.values];
+    }
+    return state;
+}
+
+const MembersModal = ({
```

Comment:
It is hard for me to follow exactly where is the cut between the two components. I understand that the previous component was getting too big and was useful to separate it, but usually the separation follows a logic of functionality. My feeling when I see this component is that the search functionality now is distributed between the two components, making it a bit more difficult to follow.

I wonder if you can give it another thought, and try to keep everything search related in the previous component, and keep in this component as much as possible of the "render related".

(Unless there is other motivation I am missing to separate these two components)